### PR TITLE
when window size is changed, use resizer to change widget size

### PR DESCRIPTION
**Pull request by origel**
I want add resize function. when window size is changed, use resizer to change widget size.

but when I add this code, in window.v:
`fn (w &ui.Window) resize(){
    w.resizer.resize(w.children)
}`

the code will cause error:
$ v run examples/users.v 
warning: examples/users.v:51:2: reference field `App*.window` must be initialized
warning: examples/users.v:146:5: reference field `ui.PictureConfig.ref` must be initialized
s end of non-void function [-Wreturn-type]
  }
  ^V error: C error. This should never happen. 
Please create a GitHub issue: https://github.com/vlang/v/issues/new/choose

when I use code:
`fn (w &ui.Window) resize(){
    //w.resizer.resize(w.children)
}`

the error is disappeared.

Can you help me find the reason? thanks.

### CODE REVIEW

**Comment by origel** [comment]
@thecodrr  I have changed the implement way. The effection just like examples/fill_resizer_h.png and examples/fill_resizer_v.png.  I know you means resize event will happen at the Window level, my resizer function is resize all widgets when needed(now I executed it when create a window), if window level resize event happen, just call resize method, all widgets size will adjust automatic.

**Comment by origel** [comment]
@thecodrr I have add row layout resizer, and the effection is row_resizer_v.png and row_resizer_h.png . Any questions, give me a feedback. thanks.

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,5 @@ examples/transitions
 examples/users
 examples/switch
 examples/message
+examples/fillresizer
+.*
```
Comment (spytheman) [inline .gitignore]: Why is this needed?

Comment (spytheman) [reply .gitignore]: I mean .* will ignore all files/folders starting with a '.' , but why?

Comment (spytheman) [reply .gitignore]: Also `examples/fillresizer` is a linux binary and should not be in this PR.

Comment (origel) [reply .gitignore]: in my project has a .project file, and I just want ignore it.  Now I have remove .* from .gitignore file. examples/fillresizer I also remove from pr.  thanks for you comments.

**Comment by origel** [comment]
@medvednikov can you merge this pr?

**Comment by medvednikov** [comment]
@thecodrr what do you think?

**Comment by thecodrr** [comment]
I will test it out in a bit. However, I don't think resizing should be done this way because the layout won't be responsive. I will look into this further.

**Comment by thecodrr** [comment]
There are a few issues with this:
1. Every layout has a different way of handling resizing. Hence, there will need to be a `resize()` method in the `ILayouter` interface. However, that is outside the scope of this PR.
2. This is not a resizing implementation, as you will be hard pressed to use it with any layout (`Row` or `Column`). @origel has basically implemented different layouts (row and fill with wrapping, kind of like a WrapPanel) which is cool in itself but can't be used with the existing layout system as independent layouts.
3. This needs to be evaluated if it fits the `VUI` philosophy.

Maybe I misunderstood but from the implementation it seems inflexible with existing layouts.

@medvednikov the decision is yours.

**Comment by origel** [comment]
@thecodrr @medvednikov if it is hard to decide, I will try to re-write this part to compatible with exists Row and Column layout. More optional layouts will make development more flexible.

**Comment by origel** [comment]
In my opinion, every Container (container is like window or group widget) can have a layout strategy. This strategy can be specified via set_layout (). Container itself is not a Layout. Layout can have different implementations, such as Row Column, filllayout, rowlayout, gridlayout, etc.

**Comment by thecodrr** [comment]
@origel Why? Just use the Layout directly and put the widgets as children. I did first think of creating a `View` widget but it seemed pointless because it would be doing the same thing as the Layout widgets itself.

**Comment by origel** [comment]
I see. You means Layout implements are special widgets which can has childern, and itself usually without ui. This way also can resolve layout problem perfectly.

If I want to implement a fill layout in my code, I just add a widget which fill function just like Stack. Right?

**Comment by origel** [comment]
These are two different design ideas for solving layout problems. The key difference in design is whether the layout is a widget or a strategy? I thought about it, and the design looks good, although I haven't used this way before.

I will modify the code to be consistent with your design.
